src/rest-server/docs/swagger.yaml

```diff
@@ -1871,52 +1618,90 @@ components:
           type: string
           enum: ['RUNNING', 'STOPPED', 'DRAINING']
           description: RUNNING -> vc is enabled, STOPPED -> vc is disabled, without either new job or running job, DRAINING -> intermedia state from RUNNING to STOPPED, in waiting on existing job.
-    Storage:
+    StorageSummary:
       type: object
       properties:
-        spn:
-          type: string
-        type:
-          type: string
-        data:
-          type: object
-          properties:
-            spn:
-              type: string
-            type:
-              type: string
-            address:
-              type: string
-            rootPath:
-              type: string
-        extension:
-          type: object
-      required: [spn, type, data, extension]
-    StorageConfig:
+        storages:
+          type: array
+          items:
+            type: object
+            properties:
+              name:
+                type: string
+              share:
+                type: boolean
+              volumeName:
+                type: string
+            required: [name, share, volumeName]
+      required: [storages]
+    StorageDetail:
       type: object
       properties:
         name:
           type: string
-        default:
+        share:
           type: boolean
-        servers:
-          type: array
-          items:
-            type: string
-        mountInfos:
-          type: array
-          items:
-            type: object
+        volumeName:
+          type: string
+        type:
+          type: string
+          enum:
+          - nfs
+          - samba
+          - azureFile
+          - azureBlob
+          - other
+          - unknown
+        data:
+          oneOf:
+          - type: object
+            description: nfs type
             properties:
-              mountPoint:
+              server:
                 type: string
               path:
                 type: string
-              server:
+            required: [server, path]
+          - type: object
+            description: samba type
+            properties:
+              address:
+                type: string
+              username:
+                type: string
+              password:
+                type: string
+            required: [address]
+          - type: object
+            description: azureFile type
+            properties:
+              shareName:
+                type: string
+              accountName:
                 type: string
-              permission:
+              accountKey:
                 type: string
-      required: [name, default, servers, mountInfos]
+            required: [shareName]
+          - type: object
+            description: azureBlob type
+            properties:
+              containerName:
+                type: string
+              accountName:
+                type: string
+              accountKey:
```

Comment:
Since there are some concerns about exposing `accountKey` to end user, consider pass `shared access signature token` instead? (both of them may be optional, and let admin to choose what to do?)

Reply:
will update according to [different types of secrets](https://github.com/Azure/kubernetes-volume-drivers/blob/master/flexvolume/blobfuse/README.md#1-create-a-secret-which-stores-azure-storage-account) used by blobfuse flexvolume

this api only exposes information in back-end persistent volume, we could add an option to show data in secrets or not

Reply:
updated

Reply:
I have no more concerns here, resolved